### PR TITLE
python312Packages.cmsis-svd: 0.4 -> 0.4-unstable-2024-01-25

### PR DESCRIPTION
**Pull request by msanft**
## Description of changes

Prerequisite to get `python3Packages.greatfet` to work again.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/cmsis-svd/default.nix [hunk 1]
```diff
@@ -2,33 +2,43 @@
...
 
   src = fetchFromGitHub {
-    owner = "posborne";
-    repo = pname;
-    rev = "python-${version}";
-    sha256 = "01f2z01gqgx0risqnbrlaqj49fmly30zbwsf7rr465ggnl2c04r0";
+    owner = "cmsis-svd";
+    repo = "cmsis-svd";
+    rev = "38d21d30abd0d4c2f34fd79d83b34392ed4bb7a3";
+    sha256 = "sha256-lFA0sNHVj4a4+EwOTmFUbM/nhmzJ4mx4GvT6Ykutakk=";
```
Comment (kirillrdy) [inline pkgs/development/python-modules/cmsis-svd/default.nix:19]:
```suggestion
    hash = "sha256-lFA0sNHVj4a4+EwOTmFUbM/nhmzJ4mx4GvT6Ykutakk=";
```

**Comment by msanft** [comment]
Result of `nixpkgs-review pr 330123` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by kirillrdy** [comment]
the reason you get 0 rebuilds is because commit message is wrong, it should be using `python3Packages.cmsis-svd` attribute

**Comment by msanft** [comment]
Result of `nixpkgs-review pr 330123` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by kirillrdy** [comment]
> > the reason you get 0 rebuilds is because commit message is wrong, it should be using `python3Packages.cmsis-svd` attribute
> 
> Interesting, thanks!

ok so i am not sure why there are still no rebuilds, still investigaging, but commit message is more correct, as its what ofborg will use.

you can also use python312Packages.cmsis-svd as its currently default version of python

**Comment by kirillrdy** [comment]
ok mystery solved https://github.com/NixOS/nixpkgs/pull/330205

**Comment by msanft** [comment]
Result of `nixpkgs-review pr 330123` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by msanft** [comment]
Result of `nixpkgs-review pr 330123` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.cmsis-svd</li>
    <li>python311Packages.cmsis-svd.dist</li>
    <li>python312Packages.cmsis-svd</li>
    <li>python312Packages.cmsis-svd.dist</li>
  </ul>
</details>

**Review by jollheef — APPROVED** [review]

**Pull request merged**